meta.yml

```
@@ -87,10 +85,140 @@ dependencies:
 
 documentation: |-
-  ## Caveat
 
-  The `lra`, `nra`, and `psatz` tactics are considered experimental features and
-  subject to change.
+  ## Documentation
+
+  **Caveat: the `lra`, `nra`, and `psatz` tactics are considered experimental
+  features and subject to change.**
+
+  This Coq library provides an adaptation of the
+  [`ring`, `field`](https://coq.inria.fr/refman/addendum/ring),
+  [`lra`, `nra`, and `psatz`](https://coq.inria.fr/refman/addendum/micromega)
+  tactics to the MathComp library.
+  See the Coq reference manual for the basic functionalities of these tactics.
+  The descriptions of these tactics below mainly focus on the differences
+  between ones provided by Coq and ones provided by this library, including the
+  additional features introduced by this library.
+
+  ### The `ring` tactic
+
+  The `ring` tactic solves a goal of the form `p = q :> R` representing a
+  polynomial equation. The type `R` must have a canonical `comRingType`
+  (commutative ring) or at least `comSemiRingType` (commutative semiring)
+  instance.
+  The `ring` tactic solves the equation by normalizing each side as a
+  polynomial, whose coefficients are either integers `Z` (if `R` is a
+  `comRingType`) or natural numbers `N`.
+
+  The `ring` tactic can decide the given polynomial equation modulo given
+  monomial equations. The syntax to use this feature is `ring: t_1 .. t_n` where
+  each `t_i` is a proof of equality `m_i = p_i`, `m_i` is a monomial, and `p_i`
+  is a polynomial.
+  Although the `ring` tactic supports ring homomorphisms (explained below), all
+  the monomials and polynomials `m_1, .., m_n, p_1, .., p_n, p, q` must have the
+  same type `R` for the moment.
+
+  Each tactic provided by this library has a preprocessor and supports
+  applications of (semi)ring homomorphisms and additive functions (N-module or
+  Z-module homomorphisms).
+  For example, suppose `f : S -> T` and `g : R -> S` are ring homomorphisms. The
+  preprocessor turns a ring sub-expression of the form `f (x + g (y * z))` into
+  `f x + f (g y) + f (g z)`.
+  A composition of homomorphisms from the initial objects `nat`, `N`, `int`, and
+  `Z` is automatically normalized to the canonical one. For example, if `R` in
+  the above example is `int`, the result of the preprocessing should be
+  `f x + y%:~R * z%:~R` where `f \o g : int -> T` is replaced with `intr`
+  (`_%:~R`).
+  Thanks to the preprocessor, the `ring` tactic supports the following
+  constructs apart from homomorphism applications:
+  - `GRing.zero` (`0%R`),
+  - `GRing.add` (`+%R`),
+  - `addn`,
+  - `N.add`,
+  - `Z.add`,
+  - `GRing.natmul`,
+  - `GRing.opp` (`-%R`),
+  - `Z.opp`,
+  - `Z.sub`,
+  - `intmul`,
+  - `GRing.one` (`1%R`),
+  - `GRing.mul` (`*%R`),
+  - `muln`,
+  - `N.mul`,
+  - `Z.mul`,
+  - `GRing.exp`,[^constant_exponent]
+  - `exprz`,[^constant_exponent]
+  - `expn`,[^constant_exponent]
+  - `N.pow`,[^constant_exponent]
+  - `Z.pow`,[^constant_exponent]
+  - `GRing.inv`,
```

```suggestion
  - `GRing.inv` (`_^-1`, used in notation `x / y` as `x * y^-1`),
```


but isn't it only for the `field` tactic?

---

You are right. I should remove this line.